akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -198,7 +198,8 @@ private[http] object OutgoingConnectionBlueprint {
 
       val getNextData = () ⇒ {
         waitingForMethod = false
-        pull(dataInput)
+        if (!isClosed(dataInput)) pull(dataInput)
```

Comment:
Yes, this change is correct: getNextData can be called from methodBypassInput’s `onPush()` in which case a close event may already be pending but not yet delivered to the dataInput.